zerver/tests/test_scheduled_messages.py

```diff
@@ -81,6 +89,203 @@ def test_schedule_message(self) -> None:
             timestamp_to_datetime(scheduled_delivery_timestamp),
         )
 
+    def create_scheduled_message(self) -> None:
+        content = "Test message"
+        scheduled_delivery_datetime = timezone_now() + datetime.timedelta(minutes=5)
+        scheduled_delivery_timestamp = int(scheduled_delivery_datetime.timestamp())
+        verona_stream_id = self.get_stream_id("Verona")
+        result = self.do_schedule_message(
+            "stream", verona_stream_id, content + " 1", scheduled_delivery_timestamp
+        )
+        self.assert_json_success(result)
+
+    def test_successful_deliver_stream_scheduled_message(self) -> None:
+        logger = mock.Mock()
+        # No scheduled message
+        result = try_deliver_one_scheduled_message(logger)
+        self.assertFalse(result)
+
+        self.create_scheduled_message()
+        scheduled_message = self.last_scheduled_message()
+
+        # mock current time to be greater than the scheduled time, so that the `scheduled_message` can be sent.
+        more_than_scheduled_delivery_datetime = (
+            scheduled_message.scheduled_timestamp + datetime.timedelta(minutes=1)
+        )
+        with mock.patch(
+            "zerver.actions.scheduled_messages.timezone_now",
+            return_value=more_than_scheduled_delivery_datetime,
+        ):
```

Comment:
We possibly should switch this to use the `time_machine` module, but that can certainly be a follow-up.

Reply:
Okay. Opened #25497